cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/EOGWorklist.kt

```diff
@@ -215,89 +222,86 @@ class Worklist<K : Any, N : Any, V>() {
+
+    while (worklist.isNotEmpty()) {
+        val (nextNode, state) = worklist.pop()
+
+        val (newState, expectedUpdate) = transformation(nextNode, state.duplicate(), worklist)
+        if (worklist.update(nextNode, newState) || !expectedUpdate) {
+            nextNode.nextEOG.forEach { if (it is K) worklist.push(it, newState.duplicate()) }
         }
-        return worklist.mop()
     }
+    return worklist.mop()
+}
 
-    inline fun <reified K : PropertyEdge<Node>, V> iterateEOG(
-        startEdges: List<K>,
-        startState: State<K, V>,
-        transformation:
-            (PropertyEdge<Node>, State<K, V>, Worklist<K, K, V>) -> Pair<State<K, V>, Boolean>
-    ): State<K, V> {
-        val globalState = mutableMapOf<K, State<K, V>>()
-        for (startEdge in startEdges) {
-            globalState[startEdge] = startState
-        }
-        val worklist = Worklist(globalState)
-        startEdges.forEach { worklist.push(it, startState) }
+inline fun <reified K : PropertyEdge<Node>, V> iterateEOG(
+    startEdges: List<K>,
+    startState: State<K, V>,
+    transformation:
+        (PropertyEdge<Node>, State<K, V>, Worklist<K, K, V>) -> Pair<State<K, V>, Boolean>
+): State<K, V> {
+    val globalState = mutableMapOf<K, State<K, V>>()
+    for (startEdge in startEdges) {
+        globalState[startEdge] = startState
+    }
+    val worklist = Worklist(globalState)
+    startEdges.forEach { worklist.push(it, startState) }
 
-        while (worklist.isNotEmpty()) {
-            val (nextEdge, state) = worklist.pop()
+    while (worklist.isNotEmpty()) {
+        val (nextEdge, state) = worklist.pop()
 
-            val (newState, expectedUpdate) = transformation(nextEdge, state.duplicate(), worklist)
-            if (worklist.update(nextEdge, newState) || !expectedUpdate) {
-                nextEdge.end.nextEOGEdges.forEach {
-                    if (it is K) worklist.push(it, newState.duplicate())
-                }
+        val (newState, expectedUpdate) = transformation(nextEdge, state.duplicate(), worklist)
+        if (worklist.update(nextEdge, newState) || !expectedUpdate) {
+            nextEdge.end.nextEOGEdges.forEach {
+                if (it is K) worklist.push(it, newState.duplicate())
             }
         }
-        return worklist.mop()
     }
+    return worklist.mop()
+}
 
-    inline fun <reified K : PropertyEdge<Node>, N : Node, V> iterateEOGEN(
-        startEdges: List<K>,
-        startState: State<N, V>,
-        transformation:
-            (PropertyEdge<Node>, State<N, V>, Worklist<K, N, V>) -> Pair<State<N, V>, Boolean>
-    ): State<N, V> {
-        val globalState = mutableMapOf<K, State<N, V>>()
-        for (startEdge in startEdges) {
-            globalState[startEdge] = startState
-        }
-        val worklist = Worklist(globalState)
-        startEdges.forEach { worklist.push(it, startState) }
+inline fun <reified K : PropertyEdge<Node>, N : Node, V> iterateEOGEN(
```

Comment:
What is the difference between iterateEOG and iterateEOGEN?

Reply:
I merged them into one